efficientdet/keras/utils_keras.py

```diff
@@ -0,0 +1,129 @@
+
+from typing import Text, Union
+
+import tensorflow as tf
+
+# pylint: disable=logging-format-interpolation
+from utils import BatchNormalization, TpuBatchNormalization
+
+
+class ActivationFn(tf.keras.layers.Layer):
+  def __init__(self, act_type: Text, name='activation_fn', **kwargs):
+
+    super(ActivationFn, self).__init__()
+
+    self.act_type = act_type
+
+    if act_type == 'swish':
+      self.act = tf.nn.swish
+    elif act_type == 'swish_native':
+      self.act = lambda x: x * tf.sigmoid(x)
+    elif act_type == 'relu':
+      self.act = tf.nn.relu
+    elif act_type == 'relu6':
+      self.act = tf.nn.relu6
+    else:
+      raise ValueError('Unsupported act_type {}'.format(act_type))
+
+    self.layer = tf.keras.layers.Lambda(lambda x: self.act(x), name=name)
+
+  def call(self, inputs, **kwargs):
+    # return features
+    return self.layer(inputs)
+
+  def get_config(self):
+    base_config = super(ActivationFn, self).get_config()
+
+    return {
+      **base_config,
+      'act_type': self.act_type
+    }
+
+
+class BatchNormAct(tf.keras.layers.Layer):
+  def __init__(self,
+               is_training_bn: bool,
+               act_type: Union[Text, None],
+               init_zero: bool = False,
+               data_format: Text = 'channels_last',
+               momentum: float = 0.99,
+               epsilon: float = 1e-3,
+               use_tpu: bool = False,
+               name: Text = None,
+               parent_name: Text = None
+               ):
+
+    super(BatchNormAct, self).__init__(name=parent_name)
+
+    self.act_type = act_type
+    self.training = is_training_bn
+
+    if init_zero:
+      self.gamma_initializer = tf.zeros_initializer()
+    else:
+      self.gamma_initializer = tf.ones_initializer()
+
+    if data_format == 'channels_first':
+      self.axis = 1
+    else:
+      self.axis = 3
+
+    if is_training_bn and use_tpu:
+      self.layer = TpuBatchNormalization(axis=self.axis,
+                                         momentum=momentum,
+                                         epsilon=epsilon,
+                                         center=True,
+                                         scale=True,
+                                         gamma_initializer=self.gamma_initializer,
+                                         name=f'{parent_name}/{name}')
+    else:
+      self.layer = BatchNormalization(axis=self.axis,
+                                      momentum=momentum,
+                                      epsilon=epsilon,
+                                      center=True,
+                                      scale=True,
+                                      gamma_initializer=self.gamma_initializer,
+                                      name=f'{parent_name}/{name}')
+
+    self.act = ActivationFn(act_type, name=parent_name)
+
+  def call(self, inputs, **kwargs):
+    x = self.layer.apply(inputs, training=self.training)
```

Comment:
use __call__, instead of ```apply``` and ```call```.

Reply:
Using it will change the names

Reply:
Just keep variable names same as origin one.
I believe it won't influence it.